packages/babel-traverse/src/scope/index.js

```diff
@@ -431,14 +431,20 @@ export default class Scope {
       );
     }
 
-    let helperName = "toArray";
+    let helperName;
```

Comment:
while you are tweaking this - any ideas how [this one](https://github.com/babel/babel/issues/7258) could be solved? The only thing I've came up with was to track holey-ness on bindings for simple cases & bail out for more complex ones.

Reply:
I don't think the helpers themselves would need to change for that. Looks like the core issue is that we call `.concat` on the original. Maybe we just should call `toConsumableArray` on the original data too before `.concat`?

Reply:
It's indeed unrelated to this. I've only reminded myself about the other issue while looking at this one, so thought I'd ask about it.  

> Maybe we just should call toConsumableArray on the original data too before .concat?

That's ofc a possibility - would be good to optimize it though, as i.e. this `.concat` call gets generated often for arguments rest and we could analyze there that it's safe to call `.concat` in such situation. Going to look into this issue again, hopefully soon, gonna try to figure it out to be both safe & optimized if possible.